query/src/sessions/session.rs

```diff
@@ -68,6 +79,9 @@ impl Session {
             session_mgr,
             ref_count,
             session_ctx,
+            user_manager,
```

Comment:
Is it possible to make it lazy initialize?

Since most queries may not need this...

Reply:
It is already lazy initialization, [MetaGrpcClient::try_new](https://github.com/datafuselabs/databend/blob/51b0ba2667e84e911c6b6309803713de0a1cd0f1/common/meta/grpc/src/grpc_client.rs#L112) just create a connection pool and only make real client instance when do read/write actions.

Reply:
e.g.  Is it possible to make  .try_get_kv_client() called lazy too?


```
    pub async fn create_global(conf: Config) -> Result<Arc<UserApiProvider>> {
        let client = MetaClientProvider::new(conf.meta.to_grpc_client_config())
            .try_get_kv_client()
            .await?;

        Ok(Arc::new(UserApiProvider { client }))
    }
```

i.e. make MetaGrpcClient created lazy?

Reply:
In `try_get_kv_client`, it first checks if metaservice is local embedded, and if local, it will create a local meta store instead of grpc client. The creation of local meta store is async...